packages/web/src/common/store/notifications/fetchNotifications.ts

```diff
@@ -20,106 +19,132 @@ export function* fetchNotifications(config: FetchNotificationsParams) {
     timeOffset = Math.round(new Date().getTime() / 1000), // current unix timestamp (sec)
     groupIdOffset
   } = config
-  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
   const getFeatureEnabled = yield* getContext('getFeatureEnabled')
   const remoteConfig = yield* getContext('remoteConfigInstance')
 
+  const useDiscoveryNotifications = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.DISCOVERY_NOTIFICATIONS
+  )
+
+  const discoveryNotificationsGenesisUnixTimestamp = remoteConfig.getRemoteVar(
+    IntKeys.DISCOVERY_NOTIFICATIONS_GENESIS_UNIX_TIMESTAMP
+  )
+
+  const shouldFetchNotificationFromDiscovery =
+    useDiscoveryNotifications &&
+    discoveryNotificationsGenesisUnixTimestamp &&
+    timeOffset > discoveryNotificationsGenesisUnixTimestamp
+
+  if (shouldFetchNotificationFromDiscovery) {
+    return yield* call(fetchDiscoveryNotifications, {
+      limit,
+      timeOffset,
+      groupIdOffset
+    })
+  }
+
+  return yield* call(fetchIdentityNotifications, limit, timeOffset)
+}
+
+function* fetchIdentityNotifications(limit: number, timeOffset: number) {
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+
   const withDethroned = yield* call(
     getFeatureEnabled,
     FeatureFlags.SUPPORTER_DETHRONED_ENABLED
   )
 
-  const useDiscoveryNotifications = yield* call(
+  return yield* call(audiusBackendInstance.getNotifications, {
+    limit,
+    timeOffset,
+    withDethroned
+  })
+}
+
+function* fetchDiscoveryNotifications(params: FetchNotificationsParams) {
+  const { timeOffset, groupIdOffset, limit } = params
+  const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+  const getFeatureEnabled = yield* getContext('getFeatureEnabled')
+  const remoteConfig = yield* getContext('remoteConfigInstance')
+
+  const isRepostOfRepostEnabled = yield* call(
     getFeatureEnabled,
-    FeatureFlags.DISCOVERY_NOTIFICATIONS
+    FeatureFlags.REPOST_OF_REPOST_NOTIFICATIONS
+  )
+  const isSaveOfRepostEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
+  )
+  const isTrendingPlaylistEnabled = yield* call(
+    getFeatureEnabled,
+    FeatureFlags.SAVE_OF_REPOST_NOTIFICATIONS
```

Comment:
is this checking the wrong feature flag??

Reply:
yes RIPP